sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,312 @@
+    child : Body
+        The child body of joint.
+    coordinates: dynamicsymbol, optional
+        Coordinates of joint.
+    speeds : dynamicsymbol, optional
+        Speeds of joint.
+    parent_joint_pos : Vector, optional
+        Defines the joint's point where the parent will be connected to child.
+        Default value is masscenter of Parent Body.
+    child_joint_pos : Vector, optional
+        Defines the joint's point where the child will be connected to parent.
+        Default value is masscenter of Child Body.
+    parent_axis : Vector, optional
+        Axis of parent frame which would be be aligned with child's
+        axis. Default is x axis in parent's frame.
+    child_axis : Vector, optional
+        Axis of child frame which would be aligned with parent's
+        axis. Default is x axis in child's frame.
+
+    Examples
+    =========
+
+    This is minimal working example where parent body and child body is
+    connected via PinJoint through their masscenters.
+
+        >>> from sympy.physics.mechanics import Body, PinJoint
+        >>> parent = Body('parent')
+        >>> child = Body('child')
+        >>> P = PinJoint('P', parent, child)
+        >>> P.coordinates()
+        [P_theta(t)]
+        >>> P.speeds()
+        [P_omega(t)]
+        >>> P.kdes()
+        [P_omega(t) - Derivative(P_theta(t), t)]
+
+    This is an example of chaos pendulum where we will do all kinematics
+    using PinJoints.
+
+        >>> from sympy import symbols
+        >>> from sympy.physics.mechanics import PinJoint, Body
+        >>> from sympy.physics.vector import dynamicsymbols, ReferenceFrame
+
+    Declaring the mass, frame, speeds, coordinates etc isn't necessary as `Body` and
+    `PinJoint` can declare it themselves. But we would be declaring them for better understanding.
+
+        >>> mA, mB, lA, lB, h = symbols('mA, mB, lA, lB, h')
+        >>> theta, phi, omega, alpha = dynamicsymbols('theta phi omega alpha')
+        >>> N = ReferenceFrame('N')
+        >>> A = ReferenceFrame('A')
+        >>> B = ReferenceFrame('B')
+
+    Declaring the bodies.
+
+        >>> rod = Body('rod', frame=A, mass=mA)
+        >>> plate = Body('plate', mass=mB, frame=B)
+        >>> C = Body('C', frame=N) #Ceiling
+
+    Declaring the joint position wrt rod's masscenter, other bodies are connected through their
+    masscenters.
+
+        >>> lA = (lB - h / 2) / 2 * A.z
+        >>> lC = (lB/2 + h/4) * A.z
+
+    Rod's y axis is aligned to ceiling's(C) y axis.
+    Rod's z axis is aligned to plate's z axis.
+
+        >>> J1 = PinJoint('J1', C, rod, coordinates=theta, speeds=omega, child_joint_pos=lA, parent_axis=N.y, child_axis=A.y)
+        >>> J2 = PinJoint('J2', rod, plate, coordinates=phi, speeds=alpha, parent_joint_pos=lC, parent_axis=A.z, child_axis=B.z)
+
+    We can check the kinematics now.
+
+        >>> J1.kdes()
+        [omega(t) - Derivative(theta(t), t)]
+        >>> J2.kdes()
+        [alpha(t) - Derivative(phi(t), t)]
```

Comment:
These aren't the kinematics to check. You need to check if the velocities of the mass centers and the angular velocities of the bodies are correct.

Reply:
Also check the direction cosine matrices.

Reply:
Added those tests to docstring example.

Reply:
That looks correct. As long as you've verified that it is the same results as the example in the pydy repository.